core/src/main/java/nl/nn/adapterframework/filesystem/FtpFileSystem.java

```diff
@@ -161,14 +169,13 @@ public void deleteFile(FTPFile f) throws FileSystemException {
 		}
 	}
 
-	
 	@Override
 	public boolean folderExists(String folder) throws FileSystemException {
 		boolean isDirectory = false;
 		try {
-			FTPFile[] files = ftpClient.listFiles(folder);
-			if(files.length > 1) 
-				isDirectory = true;
+			String pwd = ftpClient.printWorkingDirectory();
+			isDirectory = ftpClient.changeWorkingDirectory(pwd + "/" + folder);
+			ftpClient.changeWorkingDirectory(pwd);
```

Comment:
Ik denk dat je deze regel in een finally clause moet doen.

Reply:
Ja dat wou ik! Maar hij kan een IOExceptie gooien :(